core/src/test/scala/no/nrk/bigquery/internal/TableUpdateOperationTest.scala

```diff
@@ -437,4 +438,53 @@ class TableUpdateOperationTest extends FunSuite {
 
   }
 
+  test("updating partitionExpiration should result in update") {
+    def testTable(expiration: Option[FiniteDuration]) = BQTableDef.Table(
+      tableId,
+      BQSchema.of(a),
+      BQPartitionType.DatePartitioned(Ident("date")),
+      description = None,
+      clustering = Nil,
+      TableLabels.Empty,
+      tableOptions = TableOptions.Empty.copy(partitionExpiration = expiration)
+    )
+
+    def remote(expiration: Option[FiniteDuration]) = Some(
+      TableInfo
+        .newBuilder(
+          tableId.underlying,
+          StandardTableDefinition.newBuilder
+            .setSchema(SchemaHelper.toSchema(BQSchema.of(a)))
+            .setTimePartitioning(
+              TimePartitioning
+                .newBuilder(Type.DAY)
+                .setField("date")
+                .setExpirationMs(expiration.map(exp => Long.box(exp.toMillis)).orNull)
+                .build()
+            )
+            .build()
+        )
+        .build()
+    )
+
+    TableUpdateOperation.from(testTable(Some(1.day)), remote(None)) match {
```

Comment:
Maybe add one more test case where we remove expiration from remote?

Reply:
Good point. I amended it to the original commit.